card/lib/card/director_register.rb

```diff
@@ -50,7 +50,11 @@ def clear
       end
 
       def fetch card, opts={}
-        directors[card] ||= Card.new_director card, opts
+        return directors[card] if directors[card]
+        directors.each_key do |dir_card|
+          return dir_card.director if dir_card.name == card.name
+        end
```

Comment:
Maybe but it's very complicated to get it right. The key can change during the phases. Sometimes we start with an empty name, sometimes '+B' becomes 'A+B' or  'A' changes and so "A+B" has to change, too.  So you have to keep track of all these changes.

The reason why now the card object doesn't always work is that the integration phase is outside of the transaction. If we save a card in the integration phase that was part of the act then its card object has changed.